nchelpers/__init__.py

```diff
@@ -120,6 +171,40 @@ class CFDataset(Dataset):
     def __init__(self, *args, **kwargs):
         super(CFDataset, self).__init__(*args, **kwargs)
 
+    def is_indirected(self, name):
+        """Return True iff the property named has an indirect value.
+        See class docstring for explanation of indirect values.
+        """
+        return _indirection_info(self.get_direct_value(name))[0]
+
+    def get_direct_value(self, name):
+        """Return the value of the named property without indirection processing.
+        See class docstring for explanation of indirect values.
+        """
+        return super(CFDataset, self).__getattribute__(name)
+
+    @prevent_infinite_recursion
+    def __getattribute__(self, name):
+        """Handle indirect values for properties.
+        See class docstring for explanation of indirect values.
+
+        :param name: (str) name of attribute
+        """
+        value = super(CFDataset, self).__getattribute__(name)  # cannot use ``getattr``, otherwise infinite recursion
+        is_indirected, indirected_property = _indirection_info(value)
+        if is_indirected:
+            # The condition for retrieving the value of an indirected property is
+            #   ``is_indirected`` and <the property named by ``indirected_property`` exists>
+            # We must test attribute existence using ``super(CFDataset, self).__getattribute__`` instead of ``hasattr``
+            # in order to process circular indirection correctly. Cannot use ``hasattr`` because it captures the
```

Comment:
The use case is "raise an exception if there is circular indirection." 

Circular indirection is when there attributes have values such as `x.foo = '@bar'` and `x.bar = '@foo'` (or any cycle, including the simplest, `x.foo = '@foo'`). There is no legitimate value for these cases, so we raise an error. 

These cases are detected by "infinite recursion" in `__getattribute__`, that is to say, when evaluation of `__getattribute__` causes it to call itself with a argument already seen during that evaluation. That's what `@prevent_infinite_recursion` detects.